spec/requests/posts_spec.rb

```diff
@@ -1,7 +1,41 @@
 require 'rails_helper'
 
 RSpec.describe 'Posts', type: :request do
-  describe 'GET /index' do
-    pending "add some examples (or delete) #{__FILE__}"
+  before do
+    @user = FactoryBot.create(:user, name: 'Masuma', posts_counter: 0)
+    @post = FactoryBot.create(:post, title: 'Sample Post', text: 'Lorem ipsum', author: @user)
+  end
+
+  describe 'GET /posts#index' do
+    it 'checks If response status was correct' do
+      get user_posts_path(user_id: @user.id, id: @post.id)
+      expect(response).to have_http_status(200)
+    end
+
+    it 'checks If a correct template was rendered' do
+      get user_posts_path(user_id: @user.id, id: @post.id)
+      expect(response).to render_template(:index)
+    end
+
+    it 'checks if the response body includes correct placeholder text' do
+      get user_posts_path(user_id: @user.id, id: @post.id)
+      expect(response.body).to include('Posts')
+    end
+  end
+  describe 'GET /posts#show' do
+    it 'checks if response status was correct' do
+      get user_posts_path(user_id: @user.id, id: @post.id)
+      expect(response).to have_http_status(200)
+    end
```

Comment:
- [ ] Same here,  This test case is supposed to check if the response status was correct for the `post#show` controller method and its corresponding route `user_post_path`, not `user_posts_path` as you did. Please get to `user_post_path` instead of `user_posts_path` for this test.